test/clib/test_ctfunc.cpp

```diff
@@ -0,0 +1,104 @@
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include <fstream>
+
+#include "cantera/core.h"
+#include "cantera/clib/ctfunc.h"
+#include "cantera/numerics/Func1.h"
+
+using namespace Cantera;
+
+TEST(ctfunc, sin)
+{
+    double omega = 2.;
+    int fcn = func_new(SinFuncType, 0, 1, &omega);
+    ASSERT_GE(fcn, 0);
+    ASSERT_EQ(func_value(fcn, 0.), 0.);
+    ASSERT_EQ(func_value(fcn, 0.5), sin(omega * 0.5));
```

Comment:
I think the equality comparisons should be done using `EXPECT_DOUBLE_EQ`. Though I'll admit the difference here is fairly minor, it's more consistent with what we should be trying to maintain elsewhere. `ASSERT_` should only be used when we need to stop the test case immediately on a failure. And truly exact floating point results are always iffy -- this relaxes the tolerance to a few units in the last place.